src/DeclarationScope.ts

```diff
@@ -332,6 +332,10 @@ export class DeclarationScope {
       this.pushReference(this.convertEntityName(node.exprName));
       return;
     }
+    if (ts.isTypeNode(node) && node.kind === ts.SyntaxKind.RestType) {
+      this.convertTypeNode((node as ts.RestTypeNode).type);
```

Comment:
Does TS have an explicit API for this, such as `ts.isRestTypeNode()`? That would be preferable to doing a manual cast here.

Reply:
There is no any in [typescript.d.ts](https://github.com/microsoft/TypeScript/search?q=RestTypeNode&unscoped_q=RestTypeNode). So I'm assuming checking property `kind` is good enough.

Reply:
Unfortunate. I filed this as an upstream feature request, since this really should be part of the upstream API: https://github.com/microsoft/TypeScript/issues/33785